client/src/app/settings/settings.component.ts

```diff
@@ -139,4 +139,20 @@ export class SettingsComponent implements OnInit {
       },
     );
   }
+  
+  public registerMagnetHandler(): void {
+    if (window.location.protocol !== "https:") {
```

Comment:
[Secure Context](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) doesn’t necessarily mean https
localhost is also considered a secure context, so it would be nice to let users who run rdt-client on the machine they browse on use this feature too.

we can use [`window.isSecureContext`](https://developer.mozilla.org/en-US/docs/Web/API/Window/isSecureContext)
```suggestion
    if (!window.isSecureContext) {
```